azuredevops/internal/service/serviceendpoint/resource_serviceendpoint_aws_terraform.go

```diff
@@ -0,0 +1,64 @@
+package serviceendpoint
+
+import (
+	"github.com/google/uuid"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/microsoft/azure-devops-go-api/azuredevops/v6/serviceendpoint"
+	"github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/utils/converter"
+	"github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/utils/tfhelper"
+)
+
+// ResourceServiceEndpointAwsForTerraform schema and implementation for "Terraform for AWS" service endpoint resource
+func ResourceServiceEndpointAwsForTerraform() *schema.Resource {
+	r := genBaseServiceEndpointResource(flattenServiceEndpointAwsForTerraform, expandServiceEndpointAwsForTerraform)
+	r.Schema["access_key_id"] = &schema.Schema{
+		Type:        schema.TypeString,
+		Required:    true,
+		DefaultFunc: schema.EnvDefaultFunc("AZDO_TF_AWS_SERVICE_CONNECTION_ACCESS_KEY_ID", nil),
+		Description: "The AWS access key ID for signing programmatic requests.",
+	}
+	r.Schema["secret_access_key"] = &schema.Schema{
+		Type:             schema.TypeString,
+		Required:         true,
+		DefaultFunc:      schema.EnvDefaultFunc("AZDO_TF_AWS_SERVICE_CONNECTION_SECRET_ACCESS_KEY", nil),
+		Description:      "The AWS secret access key for signing programmatic requests.",
+		Sensitive:        true,
+		DiffSuppressFunc: tfhelper.DiffFuncSuppressSecretChanged,
+	}
+	saSecretHashKey, saSecretHashSchema := tfhelper.GenerateSecreteMemoSchema("secret_access_key")
+	r.Schema[saSecretHashKey] = saSecretHashSchema
+	r.Schema["region"] = &schema.Schema{
+		Type:        schema.TypeString,
```

Comment:
Add validation for this property:
`ValidateFunc:     validation.StringIsNotEmpty,`